lib/rules/arrow-body-style.js

```diff
@@ -105,7 +105,7 @@ module.exports = {
 
                             const tokenAfterArrowBody = sourceCode.getTokenAfter(arrowBody);
 
-                            if (tokenAfterArrowBody && tokenAfterArrowBody.type === "Punctuator" && /^[(\[\/`+-]/.test(tokenAfterArrowBody.value)) {
+                            if (tokenAfterArrowBody && tokenAfterArrowBody.type === "Punctuator" && /^[([\/`+-]/.test(tokenAfterArrowBody.value)) {
```

Comment:
Oh, looks like you're right; I had thought `/` always needed to be escaped, but apparently it only needs to be escaped in character classes.

Sidenote: This implies that `/[/]/` is a valid regex, but oddly, `/[/]/.toString()` returns `/[\/]/` (i.e. it auto-escapes the slash, at least in Chrome). It doesn't seem to do this for other escaped characters.